proxy/src/control/destination/background.rs

```diff
@@ -550,9 +550,14 @@ fn pb_to_addr_meta(
     set_labels: &HashMap<String, String>,
 ) -> Option<(SocketAddr, Metadata)> {
     let addr = pb.addr.and_then(pb_to_sock_addr)?;
-    let label_iter = set_labels.iter().chain(pb.metric_labels.iter());
+
+    let mut labels = set_labels.iter()
+        .chain(pb.metric_labels.iter())
+        .collect::<Vec<_>>();
```

Comment:
It would be nice to not have to `collect` this and then call `into_iter` again, but I can't really see a way around it, and this isn't in the hot path, so it doesn't really matter...

Reply:
Sorting requires that we collect; so I think this is basically fine.

Reply:
Yeah, the only alternative I could think of was passing in the the labels as an already-sorted structure like `BTreeMap`, but `prost` provides them to us as `HashMap`s, so we can't really do that here...